integrations/spark/spark-3.1/openhouse-spark-itest/src/test/java/com/linkedin/openhouse/spark/e2e/extensions/SetSnapshotsRetentionPolicyTest.java

```diff
@@ -0,0 +1,229 @@
+    mockTableService.enqueue(mockResponse(201, tableAfterSetSnapshotsRetention)); // doCommit()
+    String ddlWithSchema =
+        "ALTER TABLE openhouse.dbSetSnapshotsRetention.t1 SET POLICY (SNAPSHOT_RETENTION TIME=24H COUNT=10)";
+    Assertions.assertDoesNotThrow(() -> spark.sql(ddlWithSchema));
+  }
+
+  @Test
+  public void testSetSnapshotsRetentionPolicyTimeOrCount() {
+    Object existingTable =
+        mockGetTableResponseBody(
+            "dbSetSnapshotsRetention",
+            "t2",
+            "c1",
+            "dbSetSnapshotsRetention.t2",
+            "u1",
+            mockTableLocation(
+                TableIdentifier.of("dbSetSnapshotsRetention", "t2"),
+                convertSchemaToDDLComponent(baseSchema),
+                ""),
+            "V1",
+            baseSchema,
+            null,
+            null);
+    Object existingOpenhouseTable =
+        mockGetTableResponseBody(
+            "dbSetSnapshotsRetention",
+            "t2",
+            "c1",
+            "dbSetSnapshotsRetention.t2",
+            "u1",
+            mockTableLocationAfterOperation(
+                TableIdentifier.of("dbSetSnapshotsRetention", "t2"),
+                "ALTER TABLE %t SET TBLPROPERTIES ('openhouse.tableId'='t2')"),
+            "V1",
+            baseSchema,
+            null,
+            null);
+    Object tableAfterSetSnapshotsRetention =
+        mockGetTableResponseBody(
+            "dbSetSnapshotsRetention",
+            "t1",
+            "c1",
+            "dbSetSnapshotsRetention.t2",
+            "u1",
+            mockTableLocationAfterOperation(
+                TableIdentifier.of("dbSetSnapshotsRetention", "t2"),
+                "ALTER TABLE %t SET TBLPROPERTIES('policies'='{\"snapshotsRetention\":{\"timeCount\":24,\"granularity\":"
+                    + "\"HOUR\",\"count\":10 }}')"),
+            "V2",
+            baseSchema,
+            null,
+            null);
+    mockTableService.enqueue(mockResponse(200, existingOpenhouseTable)); // doRefresh()
+    mockTableService.enqueue(mockResponse(200, existingOpenhouseTable)); // doRefresh()
+    mockTableService.enqueue(mockResponse(201, tableAfterSetSnapshotsRetention)); // doCommit()
+    String ddlWithSchema =
+        "ALTER TABLE openhouse.dbSetSnapshotsRetention.t2 SET POLICY (VERSIONS TIME=24H COUNT=10)";
+    Assertions.assertDoesNotThrow(() -> spark.sql(ddlWithSchema));
+  }
+
+  @Test
+  public void testSetSnapshotsRetentionPolicyTimeOnly() {
+    Object existingTable =
+        mockGetTableResponseBody(
+            "dbSetSnapshotsRetention",
+            "t3",
+            "c1",
+            "dbSetSnapshotsRetention.t3",
+            "u1",
+            mockTableLocation(
+                TableIdentifier.of("dbSetSnapshotsRetention", "t3"),
+                convertSchemaToDDLComponent(baseSchema),
+                ""),
+            "V1",
+            baseSchema,
+            null,
+            null);
```

Comment:
As this is repeated multiple times can we initialize the table in separate method?

Reply:
Moved tests to `SetHistoryPolicyStatementTest` which doesn't need this bootstrap